src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java

```diff
@@ -630,19 +633,13 @@ public boolean getShowGitToolOptions() {
         }
 
         public List<SCMSourceTrait> getTraitsDefaults() {
-            // TODO: Replace with our own branch discovery implementation.
-            return Collections.singletonList(new BranchDiscoveryTrait());
+            return Collections.singletonList(new BitbucketBranchDiscoveryTrait());
```

Comment:
No need to change, but you can also go `List.of(` instead of the `Collections.singletonList`